test/test_utils.py

```diff
@@ -957,7 +960,16 @@ def test_detect_encoding(self):
                 f = NamedTemporaryFile(delete=False)
                 f.write(bom)
                 f.close()
-                self.assertEqual(expected_encoding, detect_unicode_encoding(f.name))
+                self.assertEqual(expected_encoding, detect_file_encoding(f.name))
             finally:
                 f.close()
                 os.remove(f.name)
+
+    def test_detect_file_encoding_files(self):
+        test_files = {
+            get_test_data_path('eac-utf16le.log'): 'utf-16-le',
+            get_test_data_path('eac-windows1251.log'): 'windows-1251'
+        }
+
+        for file_path, expected_encoding in test_files.items():
```

Comment:
Hmmm, after some thoughts, we could even split this in two test functions.

Reply:
Got it, just wondering (as a beginner programmer) -  why 2 separate functions preferred over 1 ? considering more files might get added in the future.

Reply:
@ShubhamBhut mainly because each test function can be executed independently, and it's easier to disable a test that is failing on one platform.

We often use loops within test functions, but it somehow goes against the principle of *unit* testing.
In this specific case, we only have 2 elements in the loop, so making 2 separate functions isn't not much a problem.

We could improve that (and still use loops) by using a feature that appeared in Python 3.4: subtests (see https://docs.python.org/3.4/library/unittest.html?highlight=subtest#distinguishing-test-iterations-using-subtests).